src/components/organisms/UiContentArea/UiContentArea.vue

```diff
@@ -0,0 +1,293 @@
+import UiContentAreaItem from './_internal/UiContentAreaItem.vue';
+
+export interface ContentAreaItem {
+  label: string;
+  title: string;
+  name?: string;
+  [key: string]: unknown;
+}
+
+const props = defineProps({
+  modelValue: {
+    type: [
+      Array,
+      Object,
+    ] as PropType<ContentAreaItem[] | ContentAreaItem>,
+    default: () => ([]),
+  },
+  /**
+   * Use this props to set inside pages title.
+   */
+  title: {
+    type: String,
+    default: '',
+  },
+  /**
+   * Use this props to pass inside pages items.
+   */
+  items: {
+    type: Array as PropType<ContentAreaItem[]>,
+    default: () => ([]),
+  },
+  /**
+   * Use this props to display inside pages header.
+   */
+  hasHeader: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+   * Use this props to pass attrs for back UiButton
+   */
+  buttonBackAttrs: {
+    type: Object as PropsAttrs,
+    default: () => ({}),
+  },
+  /**
+   * Use this props to pass attrs for back button UiIcon
+   */
+  iconBackAttrs: {
+    type: Object as PropsAttrs,
+    default: () => ({ icon: 'chevron-left' }),
+  },
+  /**
+   * Use this props to pass attrs for title UiHeading
+   */
+  headingTitleAttrs: {
+    type: Object as PropsAttrs,
+    default: () => ({}),
+  },
+  /**
+   * Use this props to pass labels inside component translation.
+   */
+  translation: {
+    type: Object,
+    default: () => ({ back: 'Back to' }),
+  },
+});
+const defaultProps = computed(() => ({
+  translation: {
+    back: 'Back to',
+    ...props.translation,
+  },
+  iconBackAttrs: {
+    icon: 'chevron-left' as Icon,
+    ...props.iconBackAttrs,
+  },
+}
+));
+
+const emit = defineEmits<{(event: 'update:modelValue', value: ContentAreaItem[]): void;
+}>();
+
+const index = inject('index', 0);
+provide('index', index + 1);
+const isNested = computed(() => index > 0);
+const headerIsDisplayed = computed(() => (props.hasHeader && !isNested.value));
+
+const activeItems = inject('activeItems', computed({
+  get: () => (Array.isArray(props.modelValue)
+    ? props.modelValue
+    : [ props.modelValue ]),
+  set: (value) => { emit('update:modelValue', value); },
+}));
+provide('activeItems', activeItems);
+const sizeOfActiveItems = computed(() => (activeItems.value.length));
+const activeItem = computed<ContentAreaItem>(() => (activeItems.value[index] || {}));
+const isActive = computed(() => (Object.keys(activeItem.value).length > 0));
+const activeItemName = computed(() => activeItem.value?.name || '');
+provide('activeItemName', activeItemName);
+const currentTitle = computed(() => (activeItems.value[sizeOfActiveItems.value - 1]?.title || props.title));
+const backToTitle = computed(() => (activeItems.value[sizeOfActiveItems.value - 2]?.title || props.title));
+
+const itemsToHandle = ref({});
+provide('items', itemsToHandle);
+const itemsAsArray = computed<ContentAreaItem[]>(() => (Object.values(itemsToHandle.value)));
+const menuItems = computed(() => {
```

Comment:
Are you able to provide the type for this `computed` property, as you did on the line before? :)